packages/databyss-ui/components/Editor/slate/line/ContentEditable.js

```diff
@@ -0,0 +1,110 @@
+import React, { useRef, useEffect } from 'react'
+import { Value } from 'slate'
+import { Editor } from 'slate-react'
+import { lineStateToSlate } from './../markup'
+
+import { useEditorContext } from '../../EditorProvider'
+import hotKeys, { editorHotKeys } from './../hotKeys'
+
+import { renderMark, getBlockRanges, renderBlock } from './../slateUtils'
+
+const initalValue = node => ({
+  document: {
+    nodes: [node],
+  },
+})
+
+const SlateContentEditable = ({
+  onContentChange,
+  onEditableStateChange,
+  onDocumentChange,
+  OnToggleMark,
+  onHotKey,
+  css,
+}) => {
+  const [editorState] = useEditorContext()
+
+  const { editableState, textValue } = editorState
+
+  const editableRef = useRef(null)
+
+  // checks editor state for active block content changed
+  const checkActiveBlockContentChanged = _nextEditableState => {
+    // on first click on change returns null values for anchor block
+    if (!_nextEditableState.value.anchorBlock) {
+      return false
+    }
+    if (!textValue) {
+      const _text = _nextEditableState.value.anchorBlock.text
+      return { _text }
+    }
+
+    const _text = _nextEditableState.value.anchorBlock.text
+    if (textValue !== _text) {
+      const _ranges = getBlockRanges(_nextEditableState.value.anchorBlock)
+      return { _text, _ranges }
+    }
+    // TODO
+    // if markup is applied to selection. update state range
+    return false
+  }
+
+  const onChange = change => {
+    const { value } = change
+    if (onDocumentChange) {
+      onDocumentChange(value.document.toJSON())
+    }
+    const blockChanges = checkActiveBlockContentChanged({ value })
```

Comment:
Do we really need `checkActiveBlockContentChanged`? I know we don't want to fire `onChange` unless the text has changed, but isn't there a more efficient check for this in the single-line context? 
At the very least, we need to change the name of this fn to avoid confusion.